### PR TITLE
[#1845] appInterface now try/catch when trying to load an app + file …

### PR DESCRIPTION
**Pull request by Skylsmoi**
…is now ES5 complient

https://github.com/tracim/tracim/issues/1845
see last comments

## Checkpoints

The goal of this section is to help code reviewer to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

- [x] Code is clear enoug
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

### CODE REVIEW

**Comment by Skylsmoi** [comment]
I have rewrote almost the entire file to remove every appearance of ES6 code that existed and changed it to code ES5 complient

**Comment by Skylsmoi** [comment]
added [wip] because there are some missing fixes

File: frontend/dist/assets/tracim/appInterface.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-  let prevSelectedAppFeature = ''
-  let prevSelectedAppFullScreen = ''
-
-  getSelectedApp = name => {
-    switch (name) {
+var prevSelectedAppFeature = ''
+var prevSelectedAppFullScreen = ''
+var APP_NOT_LOADED = 'appNotLoaded'
+var TIME_TO_RETRY = 500
+var RETRY_TIMEOUT = 5000
```
Comment (lebouquetin) [inline frontend/dist/assets/tracim/appInterface.js]: what is the usage of TIME_TO_RETRY and RETRY_TIMEOUT?

Comment (Skylsmoi) [reply frontend/dist/assets/tracim/appInterface.js]: TIME_TO_RETRY is time to wait until the next try to open the app

File: frontend/dist/assets/tracim/appInterface.js [hunk 2]
```diff
@@ -21,92 +28,143 @@
...
+    case 'hide_popupCreateWorkspace':
+      console.log('%cGLOBAL_eventReducer Custom Event', 'color: #28a745', type, data)
+      getSelectedApp(data.name).unmountApp('popupCreateContentContainer')
+      break
+    case 'unmount_app':
+      console.log('%cGLOBAL_eventReducer Custom Event', 'color: #28a745', type, data)
+
+      var selectedApp
+
+      if (prevSelectedAppFeature !== '') {
```
Comment (lebouquetin) [inline frontend/dist/assets/tracim/appInterface.js]: what is `prevSelectedAppFeature`? A global var? What does `prev` stand for? previous?

File: frontend/dist/assets/tracim/appInterface.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
-  getSelectedApp = name => {
-    switch (name) {
+var prevSelectedAppFeature = ''
+var prevSelectedAppFullScreen = ''
+var APP_NOT_LOADED = 'appNotLoaded'
+var TIME_TO_RETRY = 500
+var RETRY_TIMEOUT = 5000
+
+var getSelectedApp = function (appName) {
+  // FIXME - CH - 2019-06-18 - The try/catch is a temporary solution to solve the frontend, apps and appInterface.js
```
Comment (lebouquetin) [inline frontend/dist/assets/tracim/appInterface.js]: can we talk about this?

File: frontend/dist/assets/tracim/appInterface.js (hunk 1 above)
Comment (lebouquetin) [reply frontend/dist/assets/tracim/appInterface.js]: Could you put `60000` as `RETRY_TIMEOUT`

File: frontend/dist/assets/tracim/appInterface.js (hunk 2 above)
Comment (Skylsmoi) [reply frontend/dist/assets/tracim/appInterface.js]: "previously"
I'll rename to remove abbreviation

**Pull request merged**